kafkabp/consumer_test.go

```diff
@@ -0,0 +1,206 @@
+package kafkabp
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/Shopify/sarama"
+	"github.com/Shopify/sarama/mocks"
+)
+
+func TestNewConsumer(t *testing.T) {
+	var cfg ConsumerConfig
+
+	// Config with no Brokers should not create a new consumer and throw
+	// ErrBrokersEmpty
+	c, err := NewConsumer(cfg)
+	if c != nil {
+		t.Errorf("expected config to be nil, got %v", c)
+	}
+	if !errors.Is(err, ErrBrokersEmpty) {
+		t.Errorf("expected error %v, got %v", ErrBrokersEmpty, err)
+	}
+
+	// Config with no Topic should not create a new consumer and throw
+	// ErrTopicEmpty
+	cfg.Brokers = []string{"127.0.0.1:9090", "127.0.0.2:9090"}
+	c, err = NewConsumer(cfg)
+	if c != nil {
+		t.Errorf("expected config to be nil, got %v", c)
+	}
+	if !errors.Is(err, ErrTopicEmpty) {
+		t.Errorf("expected error %v, got %v", ErrTopicEmpty, err)
+	}
+}
+
+func TestKafkaConsumer_Consume(t *testing.T) {
+	kc := getTestMockConsumer(t)
+	pc, pc1 := setupPartitionConsumers(t, kc)
+	kMsg := getTestKafkaMessage("key1", "value1")
+	kMsg1 := getTestKafkaMessage("key2", "value2")
+	kErr := errors.New("test error")
+	pc.YieldMessage(kMsg)
+	pc.YieldError(kErr)
+	pc1.YieldMessage(kMsg1)
+	pc1.YieldError(kErr)
+
+	var consumedMsgs []*sarama.ConsumerMessage
+	var msgLock sync.Mutex
+	var consumedErrs []error
+	var errLock sync.Mutex
+
+	var wg sync.WaitGroup
+	wg.Add(4) // 2 kMsgs and 2 kerrs
+	// Use a goroutine here since kc.Consume is a blocking call
+	go func() {
+		kc.Consume(
+			func(_ context.Context, msg *sarama.ConsumerMessage) error {
+				msgLock.Lock()
+				defer msgLock.Unlock()
+				consumedMsgs = append(consumedMsgs, msg)
+				wg.Done()
+				return nil
+			},
+			func(err error) {
+				errLock.Lock()
+				defer errLock.Unlock()
+				consumedErrs = append(consumedErrs, err)
+				wg.Done()
+			},
+		)
+	}()
+
+	wg.Wait() // wait for all kafka messages and errors to be consumed
+
+	if len(consumedMsgs) != 2 {
+		t.Errorf("expected len(consumedMsgs) == 2, got %v", len(consumedMsgs))
```

Comment:
for numbers, use `%d`

```suggestion
		t.Errorf("expected len(consumedMsgs) == 2, got %d", len(consumedMsgs))

```